dhis2/capture-app
# fix: TECH-398 moves eslint-config-dhis2 to the capture app and fixes eslint errors

Hey Joakim this is the https://jira.dhis2.org/browse/TECH-398. 🏖️ 

This is the next PR where all the eslint errors is fixed.

### What is this PR about in detail.

* Fixing all eslint errors in the code base.

* Witting a command on `package` that can be run on every new PR commit (on CI/CD) or whever we would like it to run. 
  The command is 

  ```
  eslint -c .eslintrc . --quiet
  ```

  The option `quiet` will throw only erros and not the warnings.

* Remember that each time we install the packages we have to go to node_modules and fix that broken rule. This wouldnt allow us to run a command on each PR or each push programmatically since commands with starting with `eslint` would throw an error.This PR fixes that by moving the rules from the package `eslint-config-dhis2` to our repo. 

  There are two important points I wanted to stretch.

  1. One is that the rules are exactly the same still except one line in the `.eslintrc` `node: { extensions: ['.js', '.mjs'] }`. This is there because `redux-batched-actions` would complain about some `mjs` extension not being included.
  2. And second is that the versions of the packages are the same. Meaning `eslint-config-airbnb` have at the moment a version of `18.2.0` but this would cause many changes to the codebase as it is now. For first step I am using the versions that the `eslint-config-dhis2` had even though they are a bit outdated. This way the changes in the codebase are not massive. 

I believe having these dependencies into our project gives us more control over them. We will be able to update them easier than if they were in a separate package. Also it looks to me that the `eslint-config-dhis2` is not being maintained anymore which leaves us with outdated eslint rules. Feel free to raise any discussions though in case you see any pitfalls. 


### How to test?

installing all the deps and then running `yarn linter:check` should succeed with 0 errors. This means we could add it on travis and each time we push a commit on a PR that would violate a rule we would have an error. This hopefully will allow us to focus on decresing the errors we have suppressed. 

### What is next?

After this is approved I will do open a PR that does something similar with flow errors as we have discussed. 
When the PR that fixes the flow errors is also approved we can then add the checks during the CI/CD.
